core/src/main/scala/kafka/log/LogManager.scala

```diff
@@ -149,12 +151,15 @@ class LogManager(val logDirs: Array[File],
           val logRecoveryPoint = recoveryPoints.getOrElse(topicPartition, 0L)
 
           val current = new Log(logDir, config, logRecoveryPoint, scheduler, time)
-          val previous = this.logs.put(topicPartition, current)
-
-          if (previous != null) {
-            throw new IllegalArgumentException(
-              "Duplicate log directories found: %s, %s!".format(
-              current.dir.getAbsolutePath, previous.dir.getAbsolutePath))
+          if (logDir.getName.endsWith(Log.DeleteDirSuffix)) {
```

Comment:
We could actually synchronously delete logs during load right instead of waiting for the scheduled task to kick in. OTOH maybe it is better to have a consistent/single place where we delete the logs.